pennylane/ftqc/graph_state_preparation.py

```diff
@@ -0,0 +1,176 @@
+
+            dev = qml.device('default.qubit')
+
+            @qml.qnode(dev)
+            def circuit(q, qubit_ops, entanglement_ops):
+                GraphStatePrep(qubit_graph=q, qubit_ops=qubit_ops, entanglement_ops=entanglement_ops)
+                return qml.probs()
+
+            lattice = generate_lattice([2, 2], "square")
+            q = QubitGraph(lattice.graph, id="square")
+
+            qubit_ops = qml.Y
+            entanglement_ops = qml.CNOT
+
+        The resulting circuit after applying the ``GraphStatePrep`` template is:
+
+        >>> print(qml.draw(circuit, level="device")(q, qubit_ops, entanglement_ops))
+        QubitGraph<id=(0, 1), loc=[square]>: ──Y────╭X─╭●────┤  Probs
+        QubitGraph<id=(1, 0), loc=[square]>: ──Y─╭X─│──│──╭●─┤  Probs
+        QubitGraph<id=(1, 1), loc=[square]>: ──Y─│──│──╰X─╰X─┤  Probs
+        QubitGraph<id=(0, 0), loc=[square]>: ──Y─╰●─╰●───────┤  Probs
+    """
+
+    def __init__(
+        self,
+        graph: Union[nx.Graph, QubitGraph],
+        qubit_ops: Operation = qml.H,
+        entanglement_ops: Operation = qml.CZ,
+        wires: Optional[Wires] = None,
+    ):
+        self.hyperparameters["graph"] = graph
+        self.hyperparameters["qubit_ops"] = qubit_ops
+        self.hyperparameters["entanglement_ops"] = entanglement_ops
+
+        if isinstance(graph, QubitGraph):
+            if wires is not None and set(wires) != set(graph.node_labels):
+                raise ValueError("Please ensure wires objects match labels in QubitGraph")
+            super().__init__(wires=wires if wires is not None else set(graph.graph))
+        else:
+            if wires is None:
+                raise ValueError("Please ensure wires is specified.")
+            if wires is not None and set(wires) != set(graph.nodes):
+                raise ValueError("Please ensure wires objects match labels in graph")
+            super().__init__(wires=wires)
+
+    def label(
+        self,
+        decimals: Optional[int] = None,
+        base_label: Optional[str] = None,
+        cache: Optional[dict] = None,
+    ):  # pylint: disable=unused-argument
+        r"""Defines how the graph state preparation is represented in diagrams and drawings.
+
+        Args:
+            decimals: If ``None``, no parameters are included. Else, how to round
+                the parameters. Required to match general call signature. Not used.
+            base_label: overwrite the non-parameter component of the label.
+                Required to match general call signature. Not used.
+            cache: dictionary that carries information between label calls in the
+                same drawing. Required to match general call signature. Not used.
+
+        Returns:
+            str: label to use in drawings
+        """
+        return f"GraphStatePrep({self.hyperparameters['qubit_ops'](wires=0).name}, {self.hyperparameters['entanglement_ops'].name})"
```

Comment:
Since the output here is the same as that of the __repr__ can we just call that method instead?